### PR TITLE
Feature/sen 75 convert sen lp

### PR DESCRIPTION
**Pull request by thiennv-qng**

### CODE REVIEW

File: src/applications/sen_lp/view/page/components/avatar.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Avatar, Icon } from '@senswap/sen-ui'
```
Comment (thiennv-qng) [inline src/applications/sen_lp/view/page/components/avatar.tsx:1]: new code at branch duet

File: src/applications/sen_lp/view/page/components/cardStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { Card, Row, Col, Spin, Icon, Space, Typography } from '@senswap/sen-ui'
```
Comment (thiennv-qng) [inline src/applications/sen_lp/view/page/components/cardStatus.tsx:1]: new code at branch duet

File: src/applications/sen_lp/view/page/components/priceChange.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import numeral from 'numeral'
+
+import { Typography, Icon, Space } from '@senswap/sen-ui'
+
+const arrow = ({ value, size }: { value: number; size: number }) => {
+  if (value > 0)
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/components/priceChange.tsx:6]: Compact code from 6->21

File: src/applications/sen_lp/view/page/components/priceChange.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    )
+  return (
+    <Typography.Text type="warning" style={{ fontSize: size }}>
+      <Icon name="remove-circle" />
+    </Typography.Text>
+  )
+}
+
+const percentage = ({ value, size }: { value: number; size: number }) => {
+  if (value > 0)
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/components/priceChange.tsx:26]: Compact code from 26->42

File: src/applications/sen_lp/view/page/components/viewPoolButton.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+const ViewPoolButton = ({
+  infix,
+  isDropdown,
+}: {
+  infix: string
+  isDropdown?: boolean
+}) => {
+  const dispatch = useDispatch()
+  const onOpen = async () => {
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/components/viewPoolButton.tsx:15]: Why return await?

File: src/applications/sen_lp/view/page/container/poolDetail/components/prices.tsx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+import { useSenOs } from 'helpers/senos'
+
+interface TypePrice {
+  priceChange?: number
+  price?: number
+}
+
+const Price = ({ address }: { address: string }) => {
+  const [tokenA, setTokenA] = useState<TokenInfo | any>({})
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/container/poolDetail/components/prices.tsx:19]: Not use "**any**" type

File: src/applications/sen_lp/view/page/container/poolDetail/components/prices.tsx [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    const { mint_a, mint_b, reserve_a, reserve_b } = selectedPool
+    try {
+      const tokenA = await tokenProvider.findByAddress(mint_a)
+      const tokenB = await tokenProvider.findByAddress(mint_b)
+      if (tokenA) setTokenA(tokenA)
+      else setTokenA({ address: mint_a })
+      if (tokenB) setTokenB(tokenB)
+      else setTokenB({ address: mint_b })
+      //Get token price in pool (LPT)
+      const aToLpt = Swap.oracle.rake(
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/container/poolDetail/components/prices.tsx:42]: use reserve_a/reserve_b. Not use oracle.rake

File: src/applications/sen_lp/view/page/container/poolDetail/components/prices.tsx [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+          </Typography.Text>
+        </Col>
+        <Col span={24}>
+          <Row gutter={[6, 6]}>
+            <Col span={24}>
+              <Typography.Text type="secondary">In-Pool</Typography.Text>
+            </Col>
+            <Col span={24}>
+              {[tokenA, tokenB].map((tokenInfo, i) => {
+                const tokenPrice = pricesInPool
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/container/poolDetail/components/prices.tsx:96]: pricesInPool?.[tokenInfo?.address] || 0

File: src/applications/sen_lp/view/page/container/poolDetail/components/volume.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    daily,
+  } = state?.api || undefined
+
+  // status code 403 fobbiden
+  if (statusCode) return <CardNoData />
+  if (!daily || daily.length < 1) return <CardLoading />
+
+  const chartData = daily?.map(({ volume }: { volume: any }) => volume)
+  const chartLabel = daily?.map(
+    ({ time }: { time: any }) =>
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/container/poolDetail/components/volume.tsx:34]: not use any

File: src/applications/sen_lp/view/page/container/sidebar/components/card.tsx [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+  } = useSenOs()
+  const poolSelected = useSelector(
+    (state: RootStateOrAny) => state.main?.poolSelected,
+  )
+  const [tokenInfos, setTokenInfos] = useState<Array<TokenInfo | any>>()
+
+  let amount, lp
+  if (data) {
+    amount = data.amount
+    lp = utils.undecimalize(amount, 9)
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/container/sidebar/components/card.tsx:33]: what is 9?

File: src/applications/sen_lp/view/page/container/sidebar/components/poolCard.tsx [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+    senos: { pools },
+  } = useSenOs()
+  const lptAddress = Object.keys(lpts).find((key) => lpts[key].pool === address)
+  let selectedLPT: any, selectedLPTAddress: string
+  if (lptAddress) {
+    selectedLPT = lpts[lptAddress]
+    selectedLPTAddress = lptAddress
+  }
+  const amount = selectedLPT?.amount
+  const lp = utils.undecimalize(amount, 9)
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/container/sidebar/components/poolCard.tsx:39]: what is 9?

File: src/applications/sen_lp/view/page/container/sidebar/components/poolCard.tsx [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+      return setTVL(0)
+    }
+  }, [pools, selectedAddress])
+
+  useEffect(() => {
+    getTVL()
+  }, [getTVL])
+
+  const portion =
+    Math.round((Number(lp) / tvl) * Math.pow(10, 6)) / Math.pow(10, 6)
```
Comment (dmcong) [inline src/applications/sen_lp/view/page/container/sidebar/components/poolCard.tsx:60]: * Math.pow(10, 6) / * Math.pow(10, 6)???

**Pull request merged**